scripts/mitochondria_m2_wdl/MitochondriaPipelineSlim.wdl

```diff
@@ -0,0 +1,657 @@
+      ref_fai = ref_fasta_index,
+      ref_dict = ref_dict,
+      filtered_vcf = InitialFilter.filtered_vcf,
+      gatk_override = gatk_override,
+      gatk_docker_override = gatk_docker_override,
+      preemptible_tries = preemptible_tries
+  }
+
+  call GetContamination {
+    input:
+      input_vcf = SplitMultiAllelicsAndRemoveNonPassSites.vcf_for_haplochecker,
+      preemptible_tries = preemptible_tries
+  }
+
+  call Filter as FilterContamination {
+    input:
+      raw_vcf = InitialFilter.filtered_vcf,
+      raw_vcf_index = InitialFilter.filtered_vcf_idx,
+      raw_vcf_stats = CallMt.stats,
+      run_contamination = true,
+      hasContamination = GetContamination.hasContamination,
+      contamination_major = GetContamination.major_level,
+      contamination_minor = GetContamination.minor_level,
+      verifyBamID = verifyBamID,
+      base_name = base_name,
+      ref_fasta = ref_fasta,
+      ref_fai = ref_fasta_index,
+      ref_dict = ref_dict,
+      compress = compress_output_vcf,
+      gatk_override = gatk_override,
+      gatk_docker_override = gatk_docker_override,
+      m2_extra_filtering_args = m2_filter_extra_args,
+      max_alt_allele_count = 4,
+      vaf_filter_threshold = vaf_filter_threshold,
+      f_score_beta = f_score_beta,
+      artifact_prone_sites = artifact_prone_sites,
+      artifact_prone_sites_index = artifact_prone_sites_index,
+      preemptible_tries = preemptible_tries
+ }
+
+  # This is a temporary task to handle "joint calling" until Mutect2 can produce a GVCF.
+  # This proivdes coverage at each base so low coverage sites can be considered ./. rather than 0/0.
+#  call CoverageAtEveryBase {
+#    input:
+#      input_bam = SubsetBamToChrM.output_bam,
+#      input_bai = SubsetBamToChrM.output_bai,
+#      mt_interval_list = mt_interval_list,
+#      ref_fasta = ref_fasta,
+#      ref_fasta_index = ref_fasta_index,
+#      preemptible_tries = preemptible_tries
+#  }
+  
+  call SplitMultiAllelicSites {
+    input:
+      input_vcf = FilterContamination.filtered_vcf,
+      base_name = base_name,
+      ref_fasta = ref_fasta,
+      ref_fasta_index = ref_fasta_index,
+      ref_dict = ref_dict,
+      gatk_override = gatk_override,
+      gatk_docker_override = gatk_docker_override,
+      preemptible_tries = preemptible_tries
+  }
+
+  output {
+    File out_vcf = FilterContamination.filtered_vcf
+    File out_vcf_index = FilterContamination.filtered_vcf_idx
+    File split_vcf = SplitMultiAllelicSites.split_vcf
+    File split_vcf_index = SplitMultiAllelicSites.split_vcf_index
+    File coverage_metrics =  CollectWgsMetrics.metrics
+    File contamination_metrics = GetContamination.contamination_file
+    #File base_level_coverage_metrics = CoverageAtEveryBase.table
+    Int mean_coverage = CollectWgsMetrics.mean_coverage
+    String major_haplogroup = GetContamination.major_hg
+    Float contamination = FilterContamination.contamination
+  }
+}
+
+task SubsetBamToChrM {
```

Comment:
Good idea.  Some tasks like SplitMultiAllelicSites are identical between MitochondriaPipeline and MitochondriaPipelineSlim, so are you suggesting that we create a import WDL like MitochondriaPipelineSplitMultiAllelic.wdl and call that from each of the main pipelines?  I think that makes sense and makes it more maintainable in the long term (instead of code copied twice) but a little harder to read (have to follow all the imports).   Other tasks are different (eg SubsetBamToChrM which originally takes a string contig_name but in MitochondriaPipelineSlim.wdl it takes an interval_list parameter instead (which is also used in subsequent calls).

Reply:
I created a MitochondriaPipelineFilters.wdl that covers 4 shared tasks (InitialFilter, SplitMultiAllelicsAndRemoveNonPassSites, GetContamination, FilterContamination).  This seems worth doing and shows these steps are the same between MitochondriaPipeline and MitochondriaPipelineSlim.  However, in terra, the import statement is replaced by something like "import "https://api.firecloud.org/ga4gh/v1/tools/mitochondria:AlignAndCall/versions/23/plain-WDL/descriptor" as AlignAndCall"  I don't know how to (and probably don't have permission) to put my files into api.firecloud.org.  Could we have them instead import from github itself?